src/utils/url/getAbsoluteUrl.ts

```diff
@@ -2,8 +2,8 @@
  * Returns an absolute URL based on the given path.
  */
 export function getAbsoluteUrl(path: string, baseUrl?: string): string {
-  // Ignore absolute URLs.
-  if (!path.startsWith('/')) {
+  // Ignore url with any protocol or starts with *、:
+  if (/^[a-z][a-z0-9+.-]*:/.test(path) || /^[:*]/.test(path)) {
```

Comment:
As the complexity of determining the absolute URL increases, I suggest we isolate this logic into a utility function called `isAbsoluteUrl`. We can then cover that function with unit tests to ensure it behaves as expected. Would you be interested helping me with this?

Reply:
for test, i think `getAbsoluteUrl` test case can be used for `isAbsoluteUrl`

Reply:
Not really. `getAbsoluteUrl` accepts a path and ensures that the absolute path is always returned. `isAbsoluteUrl` accepts a path and returns a boolean indicating if that path _is already absolute_. Those are different requirements.

Reply:
Or did you perhaps mean just to collocate the tests in the single `getAbsoluteUrl.test` file? In that case, I'd still prefer to have a `[module].test.ts` file convention, even if it's a small utility function (absolutely no harm in that).

Reply:
ok, set a `isAbsoluteUrl.test.ts`